root/components/PartialDateInput.js

```diff
@@ -114,15 +111,15 @@ const PartialDateInput = ({
       event: SyntheticEvent<HTMLInputElement>,
       fieldName: 'year' | 'month' | 'day',
     ) => {
-      dispatch({
+      props.dispatch({
```

Comment:
Needs to be accessed via `props.` alongside `props.uncontrolled` so that Flow refines `props` correctly, which is a union type.